trader/trader_test.go

```diff
@@ -159,3 +160,63 @@ func TestIsStateSynchronized(t *testing.T) {
 		})
 	}
 }
+
+func TestShouldSendUpdateMetric(t *testing.T) {
+	now := time.Now()
+	testCases := []struct {
+		name                 string
+		start                time.Time
+		currentUpdate        time.Time
+		lastMetricUpdate     time.Time
+		wantShouldSendMetric bool
+	}{
+		{
+			name:                 "first 5 mins - refresh",
+			start:                now.Add(-4 * time.Minute),
+			currentUpdate:        now,
+			lastMetricUpdate:     now.Add(-10 * time.Second),
```

Comment:
Tighter test condition makes sense. Happy to add the border test - my thinking was that this is just testing the math functionality in Go, rather than anything unique, but I can see it as a hedge against a typo.

Reply:
Yup, that’s mostly the benefit